tests/components/mqtt/test_init.py

```diff
@@ -80,579 +75,621 @@ async def async_mock_mqtt_client(hass, config=None):
         return mock_client()
 
 
-mock_mqtt_client = threadsafe_coroutine_factory(async_mock_mqtt_client)
+@pytest.fixture
+def mqtt_client_mock(loop, hass):
+    """Fixture to mock MQTT."""
+    client = loop.run_until_complete(async_mock_mqtt_client(hass))
+    client.reset_mock()
+    return client
 
 
-# pylint: disable=invalid-name
-class TestMQTTComponent(unittest.TestCase):
-    """Test the MQTT component."""
+@pytest.fixture
+def calls():
+    """Fixture to record calls."""
+    return []
 
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        mock_mqtt_component(self.hass)
-        self.calls = []
-        self.addCleanup(self.tear_down_cleanup)
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+@pytest.fixture
+def record_calls(calls):
+    """Fixture to record calls."""
 
     @callback
-    def record_calls(self, *args):
+    def record_calls(*args):
         """Record calls."""
-        self.calls.append(args)
+        calls.append(args)
 
-    def aiohttp_client_stops_on_home_assistant_start(self):
-        """Test if client stops on HA stop."""
-        self.hass.bus.fire(EVENT_HOMEASSISTANT_STOP)
-        self.hass.block_till_done()
-        assert self.hass.data["mqtt"].async_disconnect.called
+    return record_calls
 
-    def test_publish_calls_service(self):
-        """Test the publishing of call to services."""
-        self.hass.bus.listen_once(EVENT_CALL_SERVICE, self.record_calls)
 
-        mqtt.publish(self.hass, "test-topic", "test-payload")
+async def test_mqtt_connects_on_home_assistant_mqtt_setup(hass):
+    """Test if client is connected after mqtt init on bootstrap."""
+    await async_mock_mqtt_client(hass)
+    assert hass.data["mqtt"]._mqttc.connect.call_count == 1
```

Comment:
Use the `mqtt_mock` fixture instead of accessing hass.data.

https://github.com/home-assistant/core/blob/d278dd94777e602f23dce3690158fe414042b2fe/tests/components/mqtt/conftest.py#L8